File: 2434/Saldo do Vovô.cpp
using namespace std;
#include <iostream>
#include <vector>
#include<bits/stdc++.h>
#include <algorithm>
#include <string>


int main(){
    int n, s, f;
    vector <int> saldo;
    cin >> n >> s; cin.ignore();
    saldo.push_back(s);
    
    int menor = saldo[0];
    int soma = 0;
    for (int i = 0; i < n; i++){
        cin >> f;
        saldo.push_back(f);
    }
    for (int i = 0; i < n+1 ; i++){
           soma = soma + saldo[i];
           if(soma <= menor){
               menor = soma;
           }    
    }
    
    cout << menor << endl;
}

File: 2434/Saldo do Vovô.py
n, s = map(int, input().split())


saldo = []
saldo.append(s)

for x in range(n):
    saldo.append(int(input()))
menor = saldo[0]
soma = 0
for x in range(0,len(saldo)):
    soma = soma + saldo[x]

    if soma <= menor:
        menor = soma

print(menor)

